src/main/java/com/epam/prejap/teatrees/game/Playfield.java

```diff
@@ -81,12 +76,12 @@ private boolean isValidMove(Block block, int rowOffset, int colOffset) {
     }
 
     private void hide() {
-        forEachBrick((i, j, dot) -> grid[row + i][col + j] = 0);
+        forEachBrick((i, j, dot) -> grid.setCell(row + i, col + j, 0));
```

Comment:
It seems encapsulation of the grid is broken by allowing anyone to change its internals.

Reply:
AFAIK state of the Grid cannot be changed directly now.